### DISCUSSION TITLE
Storing large (~500k) JSON blob?

### DISCUSSION THREAD

**Discussion by mporembs** [keystonejs/keystone#7582] [Questions] [answered]
_I'm a bit of a newb, so please excuse any ignorance to 'obvious' things._

I'm using a job server to pull in a large geoJSON file and trying to store it in a JSON field on a list I've created. Later on, my front end will pull the whole thing and add it to a Mapbox map. So, I don't need to query parts of this JSON or anything like that.

My mutation fails with "PayloadTooLargeError: request entity too large" when I attempt to store the blob. Is there anyway I can make this work? It seems to be Express related, not Apollo or Prisma, but I don't entirely know what I'm doing here so perhaps that's wrong. 

Should I just be storing this as a file and reading/streaming it on the front end? The site is ultimately for my personal use, so it need not scale very far.

Thanks to anyone willing to help.

**Comment by dcousens** [comment, marked as answer]
Hi @mporembs,

This is undoubtedly related to `express` and [`body-parser`](https://www.npmjs.com/package/body-parser) defaulting the parsing limits to `100kb` by default.

Understandably I wouldn't recommend increasing this for any non-personal deployment (in fact, I'd recommend _decreasing_ it),  if only for the reason that this is probably pushing the reasonable performance limitations of `JSON.parse`/`JSON.stringify`.

*But*,  when https://github.com/keystonejs/keystone/pull/7591 is merged you will be able to provide `graphql: { bodyParser: { limit: '1m' } }` in your Keystone configuration to resolve your problem :+1:
